test/dbverify.c

```diff
@@ -140,10 +156,8 @@ compare_written_database(WacomDeviceDatabase *db)
 		int nstyli;
 		const int *styli;
 
-		assert(asprintf(&path, "%s/%d-%04x-%04x.tablet", dirname,
-				libwacom_get_bustype(*device),
-				libwacom_get_vendor_id(*device),
-				libwacom_get_product_id(*device)) != -1);
+		assert(asprintf(&path, "%s/%lu.tablet", dirname,
+				hash(libwacom_get_match(*device))) != -1);
```

Comment:
we shouldn't have duplicate matches, you can just skip the hash function and use `libwacom_get_match()` as the filename.